crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -782,18 +786,32 @@ impl<'db> TypeInferenceBuilder<'db> {
                     _ => continue,
                 };
 
-                if !base_class.is_final(self.db()) {
-                    continue;
+                if is_protocol
+                    && !(base_class.is_protocol(self.db())
+                        || base_class.is_known(self.db(), KnownClass::Object))
+                {
+                    self.context.report_lint_old(
```

Comment:
Not sure if it makes sense to add new uses of `report_lint_old`?

But I guess there is some argument for keeping consistency with nearby code, and changing them all together?

cc @BurntSushi

Reply:
Yeah it's okay, I'll fix it up. Thanks for highlighting this!